src/com/ms/silverking/cloud/dht/meta/StaticDHTCreator.java

```diff
@@ -32,204 +34,254 @@
+      boolean enableMsgGroupTrace) {
+    this.zkConfig = zkConfig;
+    this.servers = servers;
+    this.replication = replication;
+    this.dhtName = dhtName;
+    this.gcName = gcName;
+    passiveNodes = "";
+    this.port = port;
+    this.nsCreationOptions = nsCreationOptions;
+    this.gridConfigDir = new File(gridConfigDir);
+    if (!this.gridConfigDir.exists()) {
+      throw new RuntimeException(gridConfigDir + " does not exist");
+    }
+    this.nsOptionsMode = nsOptionsMode;
+    this.enableMsgGroupTrace = enableMsgGroupTrace;
+  }
+
+  private void writeSKFSConfig(String skfsConfigName, File target) throws KeeperException, IOException {
+    SKFSConfigurationZK skfsConfigZk;
+    SKFSConfiguration skfsConfig;
+    com.ms.silverking.cloud.skfs.meta.MetaClient skfsMC;
+
+    skfsMC = new com.ms.silverking.cloud.skfs.meta.MetaClient(skfsConfigName, zkConfig);
+
+    skfsConfigZk = new SKFSConfigurationZK(skfsMC);
+    skfsConfig = skfsConfigZk.readFromFile(target, -1);
+    skfsConfigZk.writeToZK(skfsConfig, null);
+  }
+
+  public void createStaticDHT(UUIDBase uuid, int initialHeapSize, int maxHeapSize, String skInstanceLogBaseVar,
+      String dataBaseVar, String skfsConfigurationFile, String classVarsFile,
+      String aliasMapNameAndFile) throws IOException, KeeperException {
+    String ringName;
+    String classVarsName;
+    String dhtConfigZkPath;
+    RingCreationResults rcResults;
+    MetaClient mc;
+    String aliasMapName;
+    String aliasMapFile;
+
+    mc = new MetaClient(dhtName, zkConfig);
+
+    if (aliasMapNameAndFile != null) {
+      String[]  toks;
+
+      toks = aliasMapNameAndFile.split(":");
+      if (toks.length != 2) {
+        throw new RuntimeException("Illegal aliasMapNameAndFile: "+ aliasMapNameAndFile);
+      }
+      aliasMapName = toks[0];
+      aliasMapFile = toks[1];
+    } else {
+      aliasMapName = null;
+      aliasMapFile = null;
     }
-    
-    private void writeSKFSConfig(String skfsConfigName, File target) throws KeeperException, IOException {
-        SKFSConfigurationZK    skfsConfigZk;
-        SKFSConfiguration    skfsConfig;
-        com.ms.silverking.cloud.skfs.meta.MetaClient         skfsMC;
-        
-        skfsMC = new com.ms.silverking.cloud.skfs.meta.MetaClient(skfsConfigName, zkConfig);
-        
-        skfsConfigZk = new SKFSConfigurationZK(skfsMC);
-        skfsConfig = skfsConfigZk.readFromFile(target, -1);
-        skfsConfigZk.writeToZK(skfsConfig, null);
+
+    // Create Ring
+    ringName = ringNameBase + uuid.toString();
+    rcResults = StaticRingCreator.createStaticRing(ringName, zkConfig, servers, replication, uuid);
+
+    // Create class vars
+    ClassVars classVars;
+    Map<String, String> varsMap;
+    MetaToolOptions mto;
+
+    classVarsName = "classVars." + uuid.toString();
+    varsMap = new HashMap<>();
+    if (classVarsFile != null) {
+      Properties p;
+
+      p = new Properties();
+      p.load(new FileInputStream(classVarsFile));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when Properties.load throws an exception. Either a) close the object returned by *FileInputStream()* in a try-finally block or b) close the resource by declaring the object returned by *FileInputStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).